### DISCUSSION TITLE
Aes-*-cfb support

### DISCUSSION THREAD

**Discussion by thepyper** [shadowsocks/shadowsocks-rust#835] [Q&A] [answered]
Hi!

Digging in older versione I've seen that shadowsocks-rust used to support aes-*-cfb ciphers as well, then it's been removed after v1.8.23.. in my understanding I imagine that is to get rid of the openssl dependency.
Question: is there any plan to bring back support for there ciphers anytime, or are they just gone? :)
I'm asking because I have some compatibility issues and checking whenever I can use shadowsocks-rust or not..
Thanks!!

**Comment by AaronChen0** [comment, marked as answer]
@thepyper. If you read the [README.md](https://github.com/shadowsocks/shadowsocks-rust/blob/master/README.md) file carefully, you would know that "deprecated stream ciphers. WARN: stream ciphers are UNSAFE!". They are disabled by default because stream-cipher has been proven vulnerable to [Redirect attack](https://github.com/shadowsocks/shadowsocks-org/issues/154). 

Nonetheless, you can enable them via "cargo build --features stream-cipher".

**Reply by thepyper** [reply]
Thanks, I'll try to build this way!
And will check about the attack you mention.

**Comment by zonyitoo** [comment]
Stream ciphers have already proved to have fatal security issue, so use it at your own risk.

**IF YOUR SERVERS GOT BLOCKED, JUST DO NOT COMPLAIN ABOUT THAT SHADOWSOCKS' PROTOCOL IS PRONE TO BE DETECTED BECAUSE OF USING THE DEPRECATED STREAM PROTOCOLS.**

**Reply by thepyper** [reply]
I'll use new ciphers where possible, but I'll try to keep compatibility with what's already out there... and won't complain :)
Thanks for the info!
